File: cpp/387.first-unique-character-in-a-string.cpp
class Solution {
public:
    int solve(string &s){
	unordered_map<char, int> freq;
	for(auto i:s)freq[i]++;
	int res = -1;
	int n = s.size();
	for(int i = 0 ; i <n; i++){
	   if(freq[s[i]]==1){
	     res = i;
	     break;
	   }
	}
	return res;
    }
    int firstUniqChar(string s) {
	return solve(s);
    }
};

File: python/387.first-unique-character-in-a-string.py
class Solution:
    def solve(self, s:str) -> int:
        freq = Counter(s)
        for i in range(len(s)):
            if freq[s[i]] == 1:
                return i
        return -1

    def firstUniqChar(self, s: str) -> int:
        return self.solve(s)
        
